tests/unit/s2n_client_cert_verify_test.c

```diff
@@ -81,5 +134,52 @@ int main(int argc, char **argv)
         EXPECT_SUCCESS(s2n_cert_chain_and_key_free(chain_and_key));
     }
 
+    /*  Test: async private key operations. */
+    {
+        struct s2n_cert_chain_and_key *chain_and_key;
+        EXPECT_SUCCESS(s2n_test_cert_chain_and_key_new(&chain_and_key,
+                S2N_DEFAULT_TEST_CERT_CHAIN, S2N_DEFAULT_TEST_PRIVATE_KEY));
+
+        struct s2n_config *config;
+        EXPECT_NOT_NULL(config = s2n_config_new());
+        EXPECT_SUCCESS(s2n_config_add_cert_chain_and_key_to_store(config, chain_and_key));
+        EXPECT_SUCCESS(s2n_config_set_async_pkey_callback(config, s2n_async_pkey_store_op));
+        EXPECT_SUCCESS(s2n_config_set_unsafe_for_testing(config));
+        /* This cipher preference is set to avoid TLS 1.3. */
+        EXPECT_SUCCESS(s2n_config_set_cipher_preferences(config, "20170210"));
+
+        /* Create connection */
+        struct s2n_connection *client_conn = s2n_connection_new(S2N_CLIENT);
+        EXPECT_NOT_NULL(client_conn);
+        EXPECT_SUCCESS(s2n_connection_set_config(client_conn, config));
+        EXPECT_SUCCESS(s2n_connection_set_client_auth_type(client_conn, S2N_CERT_AUTH_REQUIRED));
+
+        struct s2n_connection *server_conn = s2n_connection_new(S2N_SERVER);
+        EXPECT_NOT_NULL(server_conn);
+        EXPECT_SUCCESS(s2n_connection_set_config(server_conn, config));
+        EXPECT_SUCCESS(s2n_connection_set_client_auth_type(server_conn, S2N_CERT_AUTH_REQUIRED));
+
+        /* Create nonblocking pipes */
+        struct s2n_test_io_pair io_pair = { 0 };
+        EXPECT_SUCCESS(s2n_io_pair_init_non_blocking(&io_pair));
+        EXPECT_SUCCESS(s2n_connection_set_io_pair(client_conn, &io_pair));
+        EXPECT_SUCCESS(s2n_connection_set_io_pair(server_conn, &io_pair));
+
+        EXPECT_SUCCESS(s2n_try_handshake_with_async_pkey_op(server_conn, client_conn));
+
```

Comment:
Actually, one more question: how do we actually know that this is hitting the async pkey logic, and not just using the certificates synchronously?

Reply:
We know it's hitting the async pkey logic due to the `s2n_async_pkey_store_op` callback. 

I could add a flag to check if the function get's called but since we set it in the config it should always be called.

Reply:
Wouldn't this test have passed before your change, bc it wouldn't have been triggering the callback but would still be able to complete the handshake? I think you need that flag.

Reply:
Pushed an update that adds a flag to make sure the callback is invoked